chsdi/views/features.py

```diff
@@ -80,16 +85,28 @@ def view_attribute_values_geojson(request):
     return _attributes(request)
 
 
+# TODO
 def _get_feature_info_for_popup(request, params, isExtended=False, isIframe=False):
     feature, vector_model = next(_get_features(params))
     layerModel = get_bod_model(params.lang)
     # TODO Remove this ugly hack
-    layer = next(get_layers_metadata_for_params(
-        params,
-        request.db.query(layerModel),
-        layerModel,
-        layerIds=[params.layerId if params.layerId != 'ch.bfs.gebaeude_wohnungs_register_preview' else 'ch.bfs.gebaeude_wohnungs_register']
-    ))
+    # This is generating many bod query
+    try:
+        layer = cache[params.lang][params.layerId]
+        if layer:
+            log.info("======== Got from cache ========")
+    except KeyError:
+        layer = next(get_layers_metadata_for_params(
+            params,
+            request.db.query(layerModel),
+            layerModel,
+            layerIds=[params.layerId if params.layerId != 'ch.bfs.gebaeude_wohnungs_register_preview' else 'ch.bfs.gebaeude_wohnungs_register']
+        ))
+        if not hasattr(cache, params.lang):
```

Comment:
I would have gone with a 
if cache.get(params.lang, None) is not None:
as I find it a little bit easier to read. 
Your approach is still completely valid.